src/main/java/com/ldtteam/structurize/items/AbstractItemWithPosSelector.java

```diff
@@ -138,4 +139,18 @@ public float getDestroySpeed(final ItemStack stack, final BlockState state)
     {
         return Float.MAX_VALUE;
     }
+
+    /**
+     * Saves the start/end coordinates on this stack.
+     * @param tool The tool stack (assumed already been validated)
+     * @param start The new start position
+     * @param end The new end position
+     */
+    public static void setBounds(@NotNull final ItemStack tool,
```

Comment:
i would make this non-static and do if(item instanceof scantool tool) tool.setbounds(stack, start, end)

Reply:
That would just make it much harder to use it.  Not helpful.

Reply:
yeah, but overrideable

Reply:
To elaborate, here is a usage case (in `ItemCommandTool`):

```java
        final ItemStack scanTool = findScanTool(player);
        if (scanTool.isEmpty()) { /* error out */ }
```

then many many lines later:

```java
            ItemScanTool.setBounds(scanTool, from, to);
            ItemScanTool.setAnchorPos(scanTool, anchor);
```

To do what you suggest would require opening a large nested block, or swapping `scanTool.isEmpty()` for the much more long-winded `!(scanTool.getItem() instanceof ItemScanTool scanToolItem)`, just to permit using `scanToolItem` further down.

Also, items are singletons anyway, so if non-static it could be called with `ModItems.scanTool.get().setBounds(...)`, but I don't think that's better than just calling the static method either.

Your suggestion does have the merit of somewhat validating that the stack probably contains a scan tool (which was I assume what you were going for), but in all places that these are called that's already been validated by other means.  Hence just stating that it's a precondition of the method in the javadoc.

----
Another usage site is more simple (in `UpdateScanToolMessage`):

```java
        if (stack.getItem() == ModItems.scanTool.get())
        {
            ItemScanTool.setBounds(stack, from, to);
        }
```

And there it could more easily do what you're suggesting, but I don't think it's better.  The one caveat is that your version potentially allows for the `ItemScanTool` to be subclassed and it would still find it; but I'm not sure why that's a thing anyone would want to do.  And indeed I think it's safer to *not* support that case until there's a good motivation for it.